python/ray/tests/test_memory_deadlock.py

```diff
@@ -68,35 +63,46 @@ def nested_tasks_parent():
 def test_deadlock_task_with_nested_task(
     ray_with_memory_monitor,
 ):
-    with pytest.raises(ray.exceptions.OutOfMemoryError) as _:
-        ray.get(nested_tasks_parent.remote())
+    with pytest.raises(
+        (ray.exceptions.RayTaskError, ray.exceptions.OutOfMemoryError)
+    ) as _:
+        bytes1 = get_additional_bytes_to_reach_memory_usage_pct(
+            memory_usage_threshold_fraction - 0.1
+        )
+        bytes2 = (
+            get_additional_bytes_to_reach_memory_usage_pct(
+                memory_usage_threshold_fraction + 0.2
+            )
+            - bytes1
+        )
+        ray.get(task_with_nested_task.remote(bytes1, bytes2, None, None))
```

Comment:
add parameters to the call? 

i.e. task_with_nested_task.remote(
task = ...
nested_task = ..
barrier = ...
instance_id = ...
)